### ISSUE TITLE
Invalid docs for Model.count() & Model.countDocuments()

### ISSUE THREAD

**Issue by sebastian-nowak** [Automattic/mongoose#6694]
`Model.count()` is now deprecated in favor of `Model.countDocuments()`, but the warning was accidentally added to the latter.

Model.count() description:
```
Counts number of matching documents in a database collection.
```

Model.countDocuments() description:
```
Counts number of matching documents in a database collection. 
This method is deprecated, use countDocuments() instead.
```

**Comment by sebastian-nowak** [comment]
Oh, and the example source code for `countDocuments` is calling `count`.

**Comment by vkarpov15** [comment]
Fixed in https://github.com/Automattic/mongoose/commit/edb91b2e922cff2bd919cb0a0698d0042f78f9d2, will be released with 5.2.2 :+1:
